lib/app_profiler/profile/vernier.rb

```diff
@@ -17,7 +17,7 @@ def format
     end
 
     def view(params = {})
-      raise NotImplementedError
+      Viewer::FirefoxRemoteViewer.view(self, **params)
```

Comment:
`AppProfiler.vernier_viewer.view(self, **params)`

Reply:
The developer should not be forced into a single viewer. Ideally, we should have a non-remote viewer that is default that opens in a web browser like the regular speedscope viewer does.

Reply:
https://github.com/Shopify/app_profiler/tree/firefox_viewer_local

Reply:
@gmcgibbon i think i misunderstand what you mean by remote / local.

There is no way to just "yarn run" firefox-profiler with a profile like you can with speedscope, it needs a web server. But we already have a web server from the app we are profiling in most cases. If by "local" you mean it includes its own web server, lets do that in another PR - this one is complex enough and has been open for long enough, and the current implementation will satisfy most users.

Reply:
Yes, I'm open to doing this in another PR but I think we still need to do this work is what I'm saying in order to close the issue I opened. The DX of the remote middleware isn't nearly as good.

Reply:
Let's apply some of the other fixes I have here though: https://github.com/Shopify/app_profiler/compare/firefox-viewer...firefox_viewer_local

It introduces a proper deprecator, and fixes some bugs in this approach.

Reply:
> Let's apply some of the other fixes I have here though: [firefox-viewer...firefox_viewer_local](https://github.com/Shopify/app_profiler/compare/firefox-viewer...firefox_viewer_local)
> 
> It introduces a proper deprecator, and fixes some bugs in this approach.

You seem to be moving much faster than me on this, wanna just push directly to this branch? Will be faster than telling me what needs to be done, I'm not a ruby or rails dev.